File: leetcode-04/solutions/cpp/0199.cpp
class Solution {
 public:
  vector<int> rightSideView(TreeNode* root) {
    vector<int> ans;

    dfs(root, 1, ans);

    return ans;
  }

 private:
  void dfs(TreeNode* root, int level, vector<int>& ans) {
    if (!root) return;
    if (level > ans.size()) ans.push_back(root->val);

    dfs(root->right, level + 1, ans);
    dfs(root->left, level + 1, ans);
  }
};

File: leetcode-04/solutions/python3/0199.py
class Solution:
    def rightSideView(self, root: TreeNode) -> List[int]:
        def dfs(root: TreeNode, level: int) -> None:
            if not root:
                return
            if level > len(ans):
                ans.append(root.val)

            dfs(root.right, level + 1)
            dfs(root.left, level + 1)

        ans = []

        dfs(root, 1)

        return ans
